luxonis_ml/nn_archive/config_building_blocks/base_models/head.py

```diff
@@ -20,51 +27,42 @@ class Head(BaseModel, ABC):
     @ivar family: Decoding family.
     @type outputs: C{Outputs}
     @ivar outputs: A configuration specifying which output names from the `outputs` block of the archive are fed into the head.
-    @type classes: list
-    @ivar classes: Names of object classes recognized by the model.
-    @type n_classes: int
-    @ivar n_classes: Number of object classes recognized by the model.
+    @type metadata: C{HeadMetadata}
+    @ivar metadata: Metadata of the parser.
     """
 
     family: str = Field(description="Decoding family.")
+    parser_name: str = Field(description="Name of the parser.")
+    metadata: HeadMetadata = Field(description="Metadata of the parser.")
     outputs: Outputs = Field(
         description="A configuration specifying which output names from the `outputs` block of the archive are fed into the head."
     )
-    classes: List[str] = Field(
-        description="Names of object classes recognized by the model."
-    )
-    n_classes: int = Field(
-        description="Number of object classes recognized by the model."
-    )
+
+    @field_validator("metadata")
+    def validate_metadata(cls, value):
+        if not isinstance(value, HeadMetadata):
```

Comment:
Can this ever happen with the `pydantic` validation already in place?

Reply:
Thanks, removed in [1031e73](https://github.com/luxonis/luxonis-ml/pull/127/commits/1031e733652d54e98130f6baf7400c621414b5eb).